### PR TITLE
add Chinese blog page that is the use case of openresty

### PR DESCRIPTION
**Pull request by lizhangyong**

### CODE REVIEW

File: v2/cn/blog-jd-page-service-based-on-resty.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+如上是整个逻辑流程，可以看到我们在Nginx这一层做了很多前置逻辑处理，以此来减少后端压力，另外我们Redis集群分机房部署，如下所示：
+
+<html>
+<img src="/images/blog/jd-page-service-02.png" width="500">
+</html>
+
+即数据会写一个主集群，然后通过主从方式把数据复制到其他机房，而各个机房读自己的集群；此处没有在各个机房做一套独立的集群来保证机房之间没有交叉访问，这样做的目的是保证数据一致性。
+
+在这套新架构中，我们可以看到Nginx+Lua已经是我们应用的一部分，我们在实际使用中，也是把它做为项目开发，做为应用进行部署。
```
Comment (agentzh) [inline v2/cn/blog-jd-page-service-based-on-resty.md]: Can we replace "Nginx + Lua" with "OpenResty" throughout the article? Thanks!

File: v2/cn/blog-jd-page-service-based-on-resty.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+# 整体架构
+
+<html>
+<img src="/images/blog/jd-page-service-01.png" width="500">
+</html>
+
+整体流程:
+
+1、请求首先进入Nginx，Nginx调用Lua进行一些前置逻辑处理，如果前置逻辑不合法直接返回；然后查询本地缓存，如果命中直接返回数据；
```
Comment (moonbingbing) [inline v2/cn/blog-jd-page-service-based-on-resty.md]: Style: need spaces around english words

File: v2/cn/blog-jd-page-service-based-on-resty.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+</html>
+
+可以看到Nginx应用和Redis集群部署在同一台机器，这样好处可以消除跨机器、跨交换机或跨机柜，甚至跨机房调用。如果本地Redis集群不命中， 还是回源到Tomcat集群进行取数据。此种方式可能受限于TCP连接数，可以考虑使用unix domain socket套接字减少本机TCP连接数。如果单机内存成为瓶颈（比如单机内存最大256GB），就需要路由机制来进行Sharding，比如按照商品尾号Sharding，Redis集群一般采用树状结构挂主从部署。
+
+## 本地缓存
+
+我们把Nginx作为应用部署，因此我们大量使用Nginx共享字典作为本地缓存，OpenResty架构中，使用HttpLuaModule模块的shared dict做本地缓存（ reload不丢失）或内存级Proxy Cache，提升缓存带来的性能并减少带宽消耗；另外我们使用一致性哈希（如商品编号/分类）做负载均衡内部对URL重写提升命中率。
+
+我们在缓存数据时采用了维度化存储缓存数据，增量获取失效缓存数据（比如10个数据，3个没命中本地缓存，只需要取这3个即可）；维度如商家信息、店铺信息、商家评分、店铺头、品牌信息、分类信息等；比如我们本地缓存30分钟，调用量减少差不多3倍。
+另外我们使用一致性哈希+本地缓存，如库存数据缓存5秒，平常命中率：本地缓存25%；分布式Redis28%；回源47%；一次普通秒杀活动命中率：本地缓存 58%；分布式Redis 15%；回源27%；而某个服务使用一致哈希后命中率提升10%；对URL按照规则重写作为缓存KEY，去随机，即页面URL不管怎么变都不要让它成为缓存不命中的因素。
```
Comment (moonbingbing) [inline v2/cn/blog-jd-page-service-based-on-resty.md]: style: add blank line

File: v2/cn/blog-jd-page-service-based-on-resty.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+## 前端业务逻辑后置
+
+前端JS应该尽可能少的业务逻辑和一些切换逻辑，因为前端JS一般推送到CDN，假设逻辑出问题了，需要更新代码上线，推送到CDN然后失效各个边缘CDN节点；或者通过版本号机制在服务端模板中修改版本号上线，这两种方式都存在效率问题，假设处理一个紧急故障用这种方式处理完了可能故障也恢复了。因此我们的观点是前端JS只拿数据展示，所有或大部分逻辑交给后端去完成，即静态资源CSS/JS CDN，动态资源JSONP；前端JS瘦身，业务逻辑后置。
+在双十一期间我们的某些服务出问题了，不能更新商品信息，此时秒杀商品需要打标处理，因此我们在服务端完成了这件事情，整个处理过程只需要几十秒就能搞定，避免了商品不能被秒杀的问题。而如果在JS中完成需要耗费非常长的时间，因为JS在客户端还有缓存时间，而且一般缓存时间非常长。
+
+## 前端接口服务端聚合
+
+商品详情页上依赖的服务众多，一个类似的服务需要请求多个不相关的服务接口，造成前端代码臃肿，判断逻辑众多；而我无法忍受这种现状，我想要的结果就是前端异步请求我的一个API，我把相关数据准备好发过去，前端直接拿到数据展示即可；所有或大部分逻辑在服务端完成而不是在客户端完成；因此我们在接入层使用Lua协程机制并发调用多个相关服务然后最后把这些服务进行了合并。
+
+比如推荐服务：最佳组合、推荐配件、优惠套装；通过http://c.3.cn/recommend?methods=accessories,suit,combination&sku=1159330&cat=6728,6740,12408&lid=1&lim=6进行请求获取聚合的数据，这样原来前端需要调用三次的接口只需要一次就能吐出所有数据。
```
Comment (moonbingbing) [inline v2/cn/blog-jd-page-service-based-on-resty.md]: Style: need spaces around URI

Comment (lizhangyong) [inline v2/cn/blog-jd-page-service-based-on-resty.md]: IF add spaces around URI, when the file translated to html page with pandoc,  encounter one problem: the URI not a string any more but a link that not expected. 
Is it OK or how to fix this?  Thanks.

**Pull request closed without merging**